--- old/src/java.desktop/share/classes/sun/java2d/marlin/ByteArrayCache.java	2015-12-05 17:45:35.150570902 +0100
+++ new/src/java.desktop/share/classes/sun/java2d/marlin/ByteArrayCache.java	2015-12-05 17:45:35.006570897 +0100
@@ -74,7 +74,7 @@
     void putDirtyArray(final byte[] array, final int length) {
         if (length != arraySize) {
             if (doChecks) {
-                System.out.println("ArrayCache: bad length = " + length);
+                MarlinUtils.logInfo("ArrayCache: bad length = " + length);
             }
             return;
         }
@@ -98,7 +98,7 @@
     {
         if (length != arraySize) {
             if (doChecks) {
-                System.out.println("ArrayCache: bad length = " + length);
+                MarlinUtils.logInfo("ArrayCache: bad length = " + length);
             }
             return;
         }
