--- old/src/java.desktop/share/classes/sun/java2d/marlin/FloatArrayCache.java	2015-12-05 17:45:35.510570915 +0100
+++ new/src/java.desktop/share/classes/sun/java2d/marlin/FloatArrayCache.java	2015-12-05 17:45:35.362570910 +0100
@@ -75,7 +75,7 @@
     void putDirtyArray(final float[] array, final int length) {
         if (length != arraySize) {
             if (doChecks) {
-                System.out.println("ArrayCache: bad length = " + length);
+                MarlinUtils.logInfo("ArrayCache: bad length = " + length);
             }
             return;
         }
@@ -99,7 +99,7 @@
     {
         if (length != arraySize) {
             if (doChecks) {
-                System.out.println("ArrayCache: bad length = " + length);
+                MarlinUtils.logInfo("ArrayCache: bad length = " + length);
             }
             return;
         }
