--- old/src/java.desktop/share/classes/sun/java2d/marlin/Renderer.java	2015-12-10 22:06:08.603013801 +0100
+++ new/src/java.desktop/share/classes/sun/java2d/marlin/Renderer.java	2015-12-10 22:06:08.443013794 +0100
@@ -629,6 +629,13 @@
         }
 
         if (edgeMinY != Float.POSITIVE_INFINITY) {
+            // if context is maked as DIRTY:
+            if (rdrCtx.dirty) {
+                // may happen if an exception if thrown in the pipeline processing:
+                // clear completely buckets arrays:
+                buckets_minY = 0;
+                buckets_maxY = boundsMaxY - boundsMinY;
+            }
             // clear used part
             if (edgeBuckets == edgeBuckets_initial) {
                 // fill only used part
