--- old/src/java.desktop/share/classes/sun/java2d/marlin/MarlinConst.java	2015-12-07 17:15:43.630234248 +0100
+++ new/src/java.desktop/share/classes/sun/java2d/marlin/MarlinConst.java	2015-12-07 17:15:43.434234240 +0100
@@ -40,6 +40,8 @@
     // log misc.Unsafe alloc/realloc/free
     static final boolean logUnsafeMalloc = enableLogs
         && MarlinProperties.isLogUnsafeMalloc();
+    // do check unsafe alignment:
+    static final boolean doCheckUnsafe = false;
 
     // do statistics
     static final boolean doStats = enableLogs && MarlinProperties.isDoStats();
