diff -Nur zulu/m02.8145055.webrev/raw_files/new/src/share/ jdk/m02.8145055.webrev/raw_files/new/src/java.desktop/share/
diff -Nur zulu/m02.8145055.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/MarlinCache.java jdk/m02.8145055.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/MarlinCache.java
--- zulu/m02.8145055.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/MarlinCache.java	2019-09-19 20:53:52.000000000 +0200
+++ jdk/m02.8145055.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/MarlinCache.java	2015-12-07 17:15:42.000000000 +0100
@@ -25,7 +25,7 @@
 
 package sun.java2d.marlin;
 
-import sun.misc.Unsafe;
+import jdk.internal.misc.Unsafe;
 
 /**
  * An object used to cache pre-rendered complex paths.

