diff -Nur zulu/m03.8144630.webrev/raw_files/new/src/ jdk/m03.8144630.webrev/raw_files/new/src/java.desktop/
diff -Nur zulu/m03.8144630.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/RendererStats.java jdk/m03.8144630.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/RendererStats.java
--- zulu/m03.8144630.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/RendererStats.java	2019-09-19 20:53:57.000000000 +0200
+++ jdk/m03.8144630.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/RendererStats.java	2015-12-03 22:32:50.000000000 +0100
@@ -34,7 +34,7 @@
 import sun.java2d.marlin.stats.Histogram;
 import sun.java2d.marlin.stats.Monitor;
 import sun.java2d.marlin.stats.StatLong;
-import sun.misc.ThreadGroupUtils;
+import sun.awt.util.ThreadGroupUtils;
 
 /**
  * This class gathers global rendering statistics for debugging purposes only

