diff -Nur zulu/m04.8144446.webrev/raw_files/new/src/ jdk/m04.8144446.webrev/raw_files/new/src/java.desktop/
diff -Nur zulu/m04.8144446.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/Renderer.java jdk/m04.8144446.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/Renderer.java
--- zulu/m04.8144446.webrev/raw_files/new/src/share/classes/sun/java2d/marlin/Renderer.java	2019-09-19 20:54:01.000000000 +0200
+++ jdk/m04.8144446.webrev/raw_files/new/src/java.desktop/share/classes/sun/java2d/marlin/Renderer.java	2015-12-10 22:06:08.000000000 +0100
@@ -28,7 +28,7 @@
 import java.util.Arrays;
 import sun.awt.geom.PathConsumer2D;
 import static sun.java2d.marlin.OffHeapArray.SIZE_INT;
-import sun.misc.Unsafe;
+import jdk.internal.misc.Unsafe;
 
 final class Renderer implements PathConsumer2D, MarlinConst {
 
@@ -40,7 +40,7 @@
     private static final int ALL_BUT_LSB = 0xfffffffe;
     private static final int ERR_STEP_MAX = 0x7fffffff; // = 2^31 - 1
 
-    private static final double POWER_2_TO_32 = 0x1.0p32;
+    private static final double POWER_2_TO_32 = FloatMath.powerOfTwoD(32);
 
     // use float to make tosubpix methods faster (no int to float conversion)
     public static final float f_SUBPIXEL_POSITIONS_X

diff -Nur zulu/m04.8144446.webrev/raw_files/new/test/ jdk/m04.8144446.webrev/raw_files/new/test/
diff -Nur zulu/m04.8144446.webrev/raw_files/new/test/sun/java2d/marlin/CrashTest.java jdk/m04.8144446.webrev/raw_files/new/test/sun/java2d/marlin/CrashTest.java
--- zulu/m04.8144446.webrev/raw_files/new/test/sun/java2d/marlin/CrashTest.java	2019-09-19 20:54:02.000000000 +0200
+++ jdk/m04.8144446.webrev/raw_files/new/test/sun/java2d/marlin/CrashTest.java	2015-12-10 22:06:08.000000000 +0100
@@ -36,7 +36,7 @@
  * @test
  * @summary Simple crash rendering test using huge GeneralPaths with the Marlin renderer
  * @run main/othervm -mx512m CrashTest
- * @ignore tests that take a long time and consumes 5Gb memory
+ * @ignore tests that take a long time
  * @run main/othervm -ms4g -mx4g CrashTest -slow
 */
 public class CrashTest {

